engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -890,4 +880,14 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  private def materializeWorkflowDescriptor(id: WorkflowId, workflowSourceFiles: WorkflowSourceFiles)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
```

Comment:
@geoffjentry @mcovarr I hope you are already aware of this, but after talking to @kshakir we came to a conclusion that this actor will no longer be singleton. So it will be created (and stopped) by whoever requests some service from it. Basically it's going back to what it was in the earlier stages of this PR (of not having this singleton). This is a little bigger change now that that there are multiple places where this needs to be modified, so just thought that you should know, and raise a flag *right now* if there's anything.